crates/red_knot_test/src/parser.rs

```diff
@@ -738,83 +811,211 @@ mod tests {
     }
 
     #[test]
-    fn invalid_config_item_no_equals() {
+    fn no_duplicate_name_files_in_test() {
         let source = dedent(
             "
-            ```py foo
+            `foo.py`:
+
+            ```py
             x = 1
             ```
+
+            `foo.py`:
+
+            ```py
+            y = 2
+            ```
             ",
         );
         let err = super::parse("file.md", &source).expect_err("Should fail to parse");
-        assert_eq!(err.to_string(), "Invalid config item `foo`.");
+        assert_eq!(
+            err.to_string(),
+            "Test `file.md` has duplicate files named `foo.py`."
+        );
     }
 
     #[test]
-    fn invalid_config_item_too_many_equals() {
+    fn no_duplicate_name_files_in_test_2() {
         let source = dedent(
             "
-            ```py foo=bar=baz
+            `test_1.py`:
+
+            ```py
             x = 1
             ```
+
+            ```py
+            y = 2
+            ```
             ",
         );
         let err = super::parse("file.md", &source).expect_err("Should fail to parse");
-        assert_eq!(err.to_string(), "Invalid config item `foo=bar=baz`.");
+        assert_eq!(
+            err.to_string(),
+            "Test `file.md` has duplicate files named `test_1.py`."
```

Comment:
Maybe we should give those autogenerated file names a more distinct name, and not just `test_#.py`? `test_1.py` is simple enough that someone might actually want to use it. Something like `mdtest_snippet_#.py` or `mdtest_{mangled_test_name}_#.py` would be much more unlikely to generate a clash, and we could forbid this pattern from being used in the explicit path name.